### PR TITLE
ISSUE-1061 Cleanup CustomProcessorInfo and related usages

### PR DESCRIPTION
**Pull request by satishd**

### CODE REVIEW

**Comment by priyank5485** [comment]
@satishd I quickly took a look at this and looks good. However, I have not tested this end to end. Unless, you have tested this I think we can wait to get it merged until after next release since its only a clean up or refactoring. What do you think? +1 otherwise if you think we still should merge it before the release

File: streams/catalog/src/main/java/com/hortonworks/streamline/streams/catalog/processor/CustomProcessorInfo.java [hunk 1]
```diff
@@ -184,22 +168,74 @@ public void setDigest (String digest) {
...
-            this.setCustomProcessorImpl(config.get(CUSTOM_PROCESSOR_IMPL));
-            this.setInputSchema(Utils.getSchemaFromConfig(config.get(INPUT_SCHEMA)));
-            this.setOutputSchema(Utils.getSchemaFromConfig(config.get(OUTPUT_SCHEMA)));
-            this.setTopologyComponentUISpecification(getCustomProcessorUISpecification(componentUISpecification));
-            this.setDigest(config.get(DIGEST));
-        }
-        return this;
+    public static CustomProcessorInfo fromTopologyComponentBundle(TopologyComponentBundle topologyComponentBundle) throws IOException {
+
+        Preconditions.checkNotNull(topologyComponentBundle, "topologyComponentBundle can not bre null");
```
Comment (priyank5485) [inline streams/catalog/src/main/java/com/hortonworks/streamline/streams/catalog/processor/CustomProcessorInfo.java]: be intsead of bre

**Comment by satishd** [comment]
@priyank5485 Thanks for reviewing. Can I merge this into master branch as the current release is over?

**Comment by priyank5485** [comment]
@satishd Sure. Thanks for refactoring this. +1

**Pull request closed without merging**